web/api/v1/api.go

```diff
@@ -641,12 +683,22 @@ func (api *API) targets(r *http.Request) apiFuncResult {
 					lastErrStr = lastErr.Error()
 				}
 
+				globalURL, err := getGlobalURL(target.URL(), api.option)
+
 				res.ActiveTargets = append(res.ActiveTargets, &Target{
-					DiscoveredLabels:   target.DiscoveredLabels().Map(),
-					Labels:             target.Labels().Map(),
-					ScrapePool:         key,
-					ScrapeURL:          target.URL().String(),
-					LastError:          lastErrStr,
+					DiscoveredLabels: target.DiscoveredLabels().Map(),
+					Labels:           target.Labels().Map(),
+					ScrapePool:       key,
+					ScrapeURL:        target.URL().String(),
+					GlobalURL:        globalURL.String(),
+					LastError: func() string {
```

Comment:
This seems like an unrelated change that shouldn't be in this PR?

Reply:
This is actually done to capture the errors from the `globalURL` as well. Without this, it might be a `go-lint` error since we are not catching the error reported by the `net` library.

Reply:
Ahhh, I see. The `LastError` field encodes scrape-related errors, so I don't think it's the right place for communicating errors related to computing the global URL (which is only relevant for UI purposes so far, and not scrape-related).

From what I can see, we pre-process both the external URL (in main.go) and the target's URL (in config.go) in such a way already that errors in `getGlobalURL()` are basically impossible (hopefully). So how about just failing the entire API call here in that case with an Internal Server Error, like in this example: https://github.com/prometheus/prometheus/blob/88af97366366fcbfff0af0f33d2da196a2afe629/web/api/v1/api.go#L1018-L1020